models/contacts.js

```diff
@@ -1,14 +1,131 @@
-// const fs = require('fs/promises')
+const { readFile, writeFile } = require("fs/promises");
+const Joi = require("joi");
+const path = require("path");
+const { v4: uuidv4 } = require("uuid");
 
-const listContacts = async () => {}
 
-const getContactById = async (contactId) => {}
+const fileName = "contacts.json";
+const contactsPath = path.join(__dirname, fileName);
 
-const removeContact = async (contactId) => {}
+const getContactsArray = async () => {
+  return await readFile(contactsPath, "utf-8")
+    .then((data) => JSON.parse(data))
+    .catch((error) => console.log(error.message));
+};
 
-const addContact = async (body) => {}
+const saveContactsArray = async (contacts) => {
+  await writeFile(contactsPath, JSON.stringify(contacts, null, 2), (error) => {
+    if (error) {
+      console.error(error.message);
+    }
+  });
```

Comment:
Staraj się nie mieszać stylów, await uzywaj z try catch raczej. 

jeżeli wykonujesz coś na przykład operację na bazie danych to w przypadku błędu możesz zrobić throw a następnie w miejscu w którym Wywołujesz tę funkcję zrobić try catch żeby ten błąd przechwycić.